test/features/extensions/extensions.go

```diff
@@ -0,0 +1,72 @@
+//go:build e2e
+
+package extensions
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/test/features/cloudnative"
+	"github.com/Dynatrace/dynatrace-operator/test/helpers"
+	componentDynakube "github.com/Dynatrace/dynatrace-operator/test/helpers/components/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/test/helpers/kubeobjects/secret"
+	"github.com/Dynatrace/dynatrace-operator/test/helpers/kubeobjects/statefulset"
+	"github.com/Dynatrace/dynatrace-operator/test/helpers/tenant"
+	"github.com/Dynatrace/dynatrace-operator/test/project"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+const (
+	agSecretName                    = "ag-ca"
+	agCertificate                   = "custom-cas/agcrt.pem"
+	agCertificateAndPrivateKey      = "custom-cas/agcrtkey.p12"
+	agCertificateAndPrivateKeyField = "server.p12"
+	customPullSecretName            = "azurecr"
+	eecImageRepo                    = "extk8sregistry.azurecr.io/eec/dynatrace-eec"
+	eecImageTag                     = "1.302.0.20240916-161445"
+)
+
+func Feature(t *testing.T) features.Feature {
+	builder := features.New("extensions-components-rollout")
+
+	secretConfig := tenant.GetSingleTenantSecret(t)
+
+	options := []componentDynakube.Option{
+		componentDynakube.WithApiUrl(secretConfig.ApiUrl),
+		componentDynakube.WithCloudNativeSpec(cloudnative.DefaultCloudNativeSpec()),
+		componentDynakube.WithActiveGateTLSSecret(agSecretName),
+		componentDynakube.WithCustomPullSecret(customPullSecretName),
+		componentDynakube.WithExtensionsEnabledSpec(true),
+		componentDynakube.WithExtensionsEECImageRefSpec(eecImageRepo, eecImageTag),
+	}
+
+	testDynakube := *componentDynakube.New(options...)
+
+	// Create customPull secret
+	customPullSecret := secret.NewDockerConfigJson(customPullSecretName, testDynakube.Namespace, secretConfig.CustomPullSecret)
+	builder.Assess("create custom pull secret", secret.Create(customPullSecret))
+
+	agCrt, _ := os.ReadFile(path.Join(project.TestDataDir(), agCertificate))
+	agP12, _ := os.ReadFile(path.Join(project.TestDataDir(), agCertificateAndPrivateKey))
```

Comment:
did you forget to mention in the description that we should also add certificates to `custom-cas/...` in our `testdata`?

also is just ignoring the error smart?
we could do an easy assert, as if the test fails fast with an obvious error, versus something gets deployed with an empty secret, that will cause a random error few minutes down the line

Reply:
it;s already there, I copied similar code from different test, have ticket to replace it when we can generate this type of certificate using go.

Reply:
@0sewa0  copied from https://github.com/dynatrace/dynatrace-operator/blob/147cc74c76b5998129d4806c129184ca1728af0f/test/features/cloudnative/codemodules/codemodules.go#L275-L285 🙈

Reply:
whoops, I should have looked deeper 😖 

thx 😘

Reply:
fixed up in bfbbfd90